src/highdicom/pr/sop.py

```diff
@@ -1572,65 +969,169 @@ def __init__(
                 raise ValueError(
-                    'For color presentation states, all referenced '
-                    'images must have three samples per pixel.'
+                    'For advanced blending presentation state, all referenced '
+                    'images must have the same Frame of Reference UID.'
                 )
+
         super().__init__(
-            referenced_images=referenced_images,
+            study_instance_uid=ref_im.StudyInstanceUID,
             series_instance_uid=series_instance_uid,
             series_number=series_number,
             sop_instance_uid=sop_instance_uid,
-            sop_class_uid=ColorSoftcopyPresentationStateStorage,
+            sop_class_uid=AdvancedBlendingPresentationStateStorage,
             instance_number=instance_number,
             manufacturer=manufacturer,
+            modality='PR',
+            patient_id=ref_im.PatientID,
+            transfer_syntax_uid=transfer_syntax_uid,
+            patient_name=ref_im.PatientName,
+            patient_birth_date=ref_im.PatientBirthDate,
+            patient_sex=ref_im.PatientSex,
+            accession_number=ref_im.AccessionNumber,
+            study_id=ref_im.StudyID,
+            study_date=ref_im.StudyDate,
+            study_time=ref_im.StudyTime,
+            referring_physician_name=getattr(
+                ref_im, 'ReferringPhysicianName', None
+            ),
+            **kwargs
+        )
+        self.copy_patient_and_study_information(ref_im)
+        self.copy_specimen_information(ref_im)
+
+        # Advanced Blending Presentation State
+        blending_input_numbers = np.zeros(
+            (len(blending_display), ),
+            dtype=int
+        )
+        ref_item = blending[0]
+        if ref_item.StudyInstanceUID != ref_im.StudyInstanceUID:
+            raise ValueError(
+                'Items of argument "blending" must have the same '
+                'Study Instance UID as the referenced images.'
+            )
+
+        for i, item in enumerate(blending):
+            if not isinstance(item, AdvancedBlending):
+                raise TypeError(
+                    'Items of argument "blending" must have type '
+                    'AdvancedBlending.'
+                )
+            if item.StudyInstanceUID != ref_item.StudyInstanceUID:
+                raise ValueError(
+                    'All items of argument "blending" must have the same '
+                    'Study Instance UID.'
+                )
+            blending_input_numbers[i] = int(item.BlendingInputNumber)
+
+        if not np.array_equal(
+            blending_input_numbers,
+            np.arange(1, len(blending_input_numbers) + 1, 1)
+        ):
+            raise ValueError(
+                'The values of attribute Blending Input Number of items of '
+                'argument "blending" must be ordinal numbers starting from 1 '
+                'and monotonically increasing by 1.'
+            )
```

Comment:
This seems quite overly complex, what's wrong with this:
```suggestion
        for i, item in enumerate(blending, 1):
            if not isinstance(item, AdvancedBlending):
                raise TypeError(
                    'Items of argument "blending" must have type '
                    'AdvancedBlending.'
                )
            if item.StudyInstanceUID != ref_item.StudyInstanceUID:
                raise ValueError(
                    'All items of argument "blending" must have the same '
                    'Study Instance UID.'
                )
            if int(item.BlendingInputNumber) != i:
                raise ValueError(
                    'The values of attribute Blending Input Number of items of '
                    'argument "blending" must be ordinal numbers starting from 1 '
                    'and monotonically increasing by 1.'
                )
```

Reply:
I think it would have to be `int(item.BlendingInputNumber) != (i + 1)`, but otherwise makes total sense.

Reply:
That is accounted for in the suggestion by starting the enumeration at 1

Reply:
Oh...I missed that. Great!